# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieService.cpp
# Commit: c067a02faf46
# Full Hash: c067a02faf4678b99e128cdc2753078a8429d51e
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2020-07-09 21:37:35
# Description:
#   Bug 1650211 - Ignore cookie settings during the shutdown, r=mayhemer
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D82498
# ==============================================================================

diff -r 378336e0a2eb -r c067a02faf46 netwerk/cookie/CookieService.cpp
--- a/netwerk/cookie/CookieService.cpp	Thu Jul 09 15:28:01 2020 +0000
+++ b/netwerk/cookie/CookieService.cpp	Thu Jul 09 14:14:41 2020 +0000
@@ -2244,7 +2244,10 @@
                                       const OriginAttributes& aAttrs,
                                       nsIURI* aHostURI, bool aFromHttp,
                                       const nsTArray<CookieStruct>& aCookies) {
-  MOZ_ASSERT(IsInitialized());
+  if (!IsInitialized()) {
+    // If we are probably shutting down, we can ignore this cookie.
+    return true;
+  }
 
   CookieStorage* storage = PickStorage(aAttrs);
   int64_t currentTimeInUsec = PR_Now();
